app/models/user.rb

```diff
@@ -9,4 +9,18 @@ class User < ApplicationRecord
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+
+  # user
+  has_many :friendships, foreign_key: 'user_id'
+  has_many :friends, class_name: 'User', through: 'friendships'
+
+  # friend
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id'
+  has_many :inverse_friends, source: 'user', through: 'inverse_friendships'
+
+  # def all_friends
+  #   friends_array = friendships.map{|friendship| friendship.friend if friendship.status}
+  #   friends_array + inverse_friendships.map{|friendship| friendship.user if friendship.status}
+  #   friends_array.compact
+  # end
```

Comment:
It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
See this [post](https://stackoverflow.com/questions/13130859/inline-comments-in-ruby/13130900#:~:text=No%2C%20Ruby%20does%20not%20have,comment%20out%20the%20one%20row), about why comments are discouraged.